### PR TITLE
Expand free school categorical eligibility to foster, homeless, migrant, and runaway children

### LINKED ISSUE #4600: Expand free school categorical eligibility to foster, homeless, migrant, Head Start, and runaway children
https://www.law.cornell.edu/cfr/text/7/245.2 says:

>Categorically eligible means considered income eligible for [free meals](https://www.law.cornell.edu/definitions/index.php?width=840&height=800&iframe=true&def_id=c583fad014426cefadd330c6ce1d824c&term_occur=999&term_src=Title:7:Subtitle:B:Chapter:II:Subchapter:A:Part:245:245.2) or [free milk](https://www.law.cornell.edu/definitions/index.php?width=840&height=800&iframe=true&def_id=231f12a8f11711464e5998f68e71cd22&term_occur=999&term_src=Title:7:Subtitle:B:Chapter:II:Subchapter:A:Part:245:245.2), as applicable, based on [documentation](https://www.law.cornell.edu/definitions/index.php?width=840&height=800&iframe=true&def_id=23f5053670118a25136a4a3d625a7e01&term_occur=999&term_src=Title:7:Subtitle:B:Chapter:II:Subchapter:A:Part:245:245.2) that a child is a member of a Family, as defined in this section, and one or more children in that [family](https://www.law.cornell.edu/definitions/index.php?width=840&height=800&iframe=true&def_id=dac9a94b47d8940de24c6570ca09f7cb&term_occur=999&term_src=Title:7:Subtitle:B:Chapter:II:Subchapter:A:Part:245:245.2) are receiving assistance under SNAP, FDPIR or the TANF program, as defined in this section. A Foster child,Homeless child, a Migrant child, a Head Start child and a Runaway child, as defined in this section, are also categorically eligible. Categorical eligibility and automatic eligibility may be used synonymously.

This follows from https://www.law.cornell.edu/uscode/text/42/1758#b_5 which provides discretionary authority for states to entitle these children.

The USC and CFR also should replace the current reference in https://github.com/PolicyEngine/policyengine-us/blob/master/policyengine_us/parameters/gov/usda/school_meals/categorical_eligibility.yaml

### PR DESCRIPTION
**Pull request by vrathi101**
Fixes #4600
Will not include Head Start child program eligibility as this will be incorporated in issue #4735

### CODE REVIEW

File: policyengine_us/variables/household/demographic/person/is_migrant_child.py
[the remark this answers is not in the document]
Comment (vrathi101) [reply policyengine_us/variables/household/demographic/person/is_migrant_child.py:1]: > we have an immigration status variable, lets use it here in a formula + child definition

I don't think the the four status we have `CITIZEN = "Citizen"
    LEGAL_PERMANENT_RESIDENT = "Legal Permanent Resident"
    DACA_TPS = (
        "Deferred Action for Childhood Arrivals or Temporary Protected Status"
    )
    UNDOCUMENTED = "Undocumented"` match the legal code definition
    
Screenshot 2024-07-05 at 12 18 00 PM

[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/variables/household/demographic/person/is_migrant_child.py:1]: I see, can we add the definitions to the documentation metadata and rename the file to `is_migratoty_child.py`

File: policyengine_us/parameters/gov/usda/school_meals/categorical_eligibility.yaml [hunk 1]
```diff
@@ -5,9 +5,16 @@ values:
...
+    - is_runaway_child
+    - is_migratory_child
+    # Also Head Start child
 
 metadata:
   unit: program
   reference:
-    - title: Extending Categorical Eligibility to Additional Children in a Household
-      href: https://www.fns.usda.gov/cn/extending-categorical-eligibility-additional-children-household
+    - title: 7 CFR § 245.2 - Definitions
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/usda/school_meals/categorical_eligibility.yaml:17]: could we keep the previous reference as well

File: policyengine_us/parameters/gov/usda/school_meals/categorical_eligibility.yaml [hunk 2]
```diff
@@ -5,9 +5,16 @@ values:
...
     - tanf # Temporary Assistance to Needy Families. Currently only available in CA and IL.
     - fdpir # Food Distribution Program on Indian Reservations.
+    - was_in_foster_care
+    - is_homeless
+    - is_runaway_child
+    - is_migratory_child
+    # Also Head Start child
 
 metadata:
   unit: program
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/usda/school_meals/categorical_eligibility.yaml:15]:
```suggestion
  unit: list
  period: year
  label: Free school meals categorical eligibility
```

File: policyengine_us/tests/policy/baseline/gov/usda/school_meals/meets_school_meal_categorical_eligibility.yaml [hunk 3]
```diff
@@ -2,12 +2,24 @@
...
+      is_runaway_child: false
+      is_migratory_child: false
   output:
       meets_school_meal_categorical_eligibility: true
 
 - name: Without SNAP, no categorical eligibility.
   period: 2022
   input:
       snap: 0
+      tanf: 0
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/usda/school_meals/meets_school_meal_categorical_eligibility.yaml:18]: could you add a couple of new test cases with true values for some of them

File: policyengine_us/variables/household/demographic/person/is_migratory_child.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from policyengine_us.model_api import *
+
+
+class is_migratory_child(Variable):
+    value_type = bool
+    entity = Person
+    label = "Is migratory child"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/demographic/person/is_migratory_child.py:7]: add reference

File: policyengine_us/variables/household/demographic/person/is_runaway_child.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from policyengine_us.model_api import *
+
+
+class is_runaway_child(Variable):
+    value_type = bool
+    entity = Person
+    label = "Is runaway child"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/demographic/person/is_runaway_child.py:7]: add reference

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/usda/school_meals/categorical_eligibility.yaml (hunk 1 above)
Comment (vrathi101) [reply policyengine_us/parameters/gov/usda/school_meals/categorical_eligibility.yaml:17]: Max mentioned to replace it in the issue, should I keep or remove?

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/usda/school_meals/categorical_eligibility.yaml:18]: lets keep then

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/usda/school_meals/categorical_eligibility.yaml [hunk 6]
```diff
@@ -5,9 +5,17 @@ values:
     - snap # Supplemental Nutrition Assistance Program.
     - tanf # Temporary Assistance to Needy Families. Currently only available in CA and IL.
     - fdpir # Food Distribution Program on Indian Reservations.
+    - was_in_foster_care
+    - is_homeless
+    - is_runaway_child
+    - is_migratory_child
+    # Also Head Start child
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/usda/school_meals/categorical_eligibility.yaml:12]: file an issue and amend the pr title if this pr isn't adding this criterion

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/usda/school_meals/categorical_eligibility.yaml (hunk 6 above)
Comment (MaxGhenis) [reply policyengine_us/parameters/gov/usda/school_meals/categorical_eligibility.yaml:12]: could you file an issue specifically for head start providing categorical eligibility for free school meals? we can do that independently of the actual head start logic, for example by including a placeholder variable for head start

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**